applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_2D2N.hpp

```diff
@@ -115,52 +120,148 @@ namespace Kratos
+         */
 		bounded_matrix<double,msLocalSize,msLocalSize> CreateElementStiffnessMatrix_Kd_mat();
+
+		/**
+         * @brief This function calculates the geometric part of the total stiffness matrix
+         */
 		bounded_matrix<double,msLocalSize,msLocalSize> CreateElementStiffnessMatrix_Kd_geo();
+
+		/**
+         * @brief This function calculates the co-rotating part of the total stiffness matrix
+         */
 		bounded_matrix<double,msElementSize,msElementSize> CreateElementStiffnessMatrix_Kr();
+
+		/**
+         * @brief This function assembles the total stiffness matrix
+         */
 		bounded_matrix<double,msElementSize,msElementSize> CreateElementStiffnessMatrix_Total();
 
-		void CalculateRightHandSideLinear(VectorType& rRightHandSideVector, MatrixType rLeftHandSideMatrix);
 
+		/**
+         * @brief This function globalizes matrices
+         * @param A The matrix to be globalized
+         */
 		void GlobalizeMatrix(Matrix &A);
+
+		/**
+         * @brief This function globalizes vectors
+         * @param A The vector to be globalized
+         */
 		void GlobalizeVector(Vector &A);
+
+		/**
+         * @brief This function calculates the modulus to 2 PI to keep the rotation angle between 0 and 2PI
+         * @param A The current angle
+         */
 		double Modulus2Pi(double A);
 
-		bounded_matrix<double,msElementSize,msElementSize> CreateRotationMatrix();
+		/**
+         * @brief This function calculates the transformation matrix to globalize/localize vectors and/or matrices
+         */
+		virtual bounded_matrix<double,msElementSize,msElementSize> CreateRotationMatrix();
+
 
+		/**
+         * @brief This function calculates the deformation parameters
+         */
 		bounded_vector<double,msLocalSize> CalculateDeformationParameters();
+
+		/**
+         * @brief This function calculates the internal forces w.r.t. the deformation parameters
+         */
 		bounded_vector<double,msLocalSize> CalculateInternalStresses_DeformationModes();
+
+		/**
+         * @brief This function calculates the "real" internal forces in a local reference frame
+         */
 		bounded_vector<double,msElementSize> ReturnElementForces_Local();
 
-	private:
 
-		bool mIsLinearElement = false;
-		bounded_vector<double,msLocalSize> DeformationForces = ZeroVector(msLocalSize);
-		Vector F_int_global = ZeroVector(msElementSize);
-		Matrix K_master = ZeroMatrix(msElementSize,msElementSize);
-		CrBeamElement2D2N() {};
+		void GetValueOnIntegrationPoints(
+			const Variable<array_1d<double, 3 > >& rVariable,
+			std::vector< array_1d<double, 3 > >& rOutput,
+			const ProcessInfo& rCurrentProcessInfo) override;
+
+		void CalculateOnIntegrationPoints(
+			const Variable<array_1d<double, 3 > >& rVariable,
+			std::vector< array_1d<double, 3 > >& rOutput,
+			const ProcessInfo& rCurrentProcessInfo) override;
+
+	private:
 
+		// stores the deformation modes
+		bounded_vector<double,msLocalSize> mDeformationForces = ZeroVector(msLocalSize);

```

Comment:
why do you need to store these vectors? you could pass them as arguements to the functions as needed.

Reply:
I need the DeformationForces from the previous step to calculate the geometric part of my stiffness matrix, thus I need to store this variable